### PR TITLE
thirdparty repo flag for API

### PR DESCRIPTION
**Pull request by MichaelMraka**
## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

**Review by Josca — APPROVED** [review]
I like it. Just couple of small proposals made.

File: evaluator/evaluate.go [hunk 1]
```diff
@@ -52,6 +53,7 @@ func configure() {
 	disableCompression := !utils.GetBoolEnvOrDefault("ENABLE_VMAAS_CALL_COMPRESSION", true)
 	enableAdvisoryAnalysis = utils.GetBoolEnvOrDefault("ENABLE_ADVISORY_ANALYSIS", true)
 	enablePackageAnalysis = utils.GetBoolEnvOrDefault("ENABLE_PACKAGE_ANALYSIS", true)
+	enableRepoAnalysis = utils.GetBoolEnvOrDefault("ENABLE_REPO_ANALYSIS", true)
```
[the remark this answers is not in the document]
Comment (MichaelMraka) [reply evaluator/evaluate.go:56]: https://github.com/RedHatInsights/e2e-deploy/pull/2791

File: evaluator/evaluate.go [hunk 2]
```diff
@@ -355,8 +379,21 @@ func loadPackages(tx *gorm.DB, accountID, systemID int,
...
 }
 
+func countRepos(tx *gorm.DB, accountID, systemID int) (int, error) {
+	var count int
+	err := tx.Table("system_repo").
+		Where("rh_account_id = ?", accountID).
+		Where("system_id = ?", systemID).
+		Count(&count).Error
+	if err != nil {
+		utils.Log("err", err.Error()).Warn("counting repos")
```
[the remark this answers is not in the document]
Comment (MichaelMraka) [reply evaluator/evaluate.go]: added account/system id logging

File: evaluator/evaluate.go [hunk 3]
```diff
@@ -151,7 +153,12 @@ func evaluateAndStore(tx *gorm.DB, system *models.SystemPlatform, vmaasData vmaa
 		return errors.Wrap(err, "Package analysis failed")
 	}
 
-	err = updateSystemPlatform(tx, system, newSystemAdvisories, installed, updatable)
+	thirdparty, err := analyzeRepos(tx, system, vmaasData)
```
[the remark this answers is not in the document]
Comment (MichaelMraka) [reply evaluator/evaluate.go]: changed

File: database_admin/migrations/060_thirdparty.down.sql [hunk 4]
```diff
@@ -0,0 +1 @@
+ALTER TABLE system_platform drop column thirdparty;
```
[the remark this answers is not in the document]
Comment (MichaelMraka) [reply database_admin/migrations/060_thirdparty.down.sql]: changed

**Review by Josca — APPROVED** [review]

**Pull request merged**